### ISSUE TITLE
Model evaluation on test set yields different results on the same model

### ISSUE THREAD

**Issue by swenkel** [jhfjhfj1/autokeras#162]
### Bug Description
Test set evaluation yields different results if re-run.

### Reproducing Steps
re-run multiple times:
```python
y_pred = clf.evaluate(X_test, y_test)
print(y_pred)
```
will yield different results:
```
0.0992
0.1032
0.101
0.0989
```

Further, using manual evaluation:
```python
y_prediction = clf.predict(x_test=X_test)
from sklearn.metrics import accuracy_score
accuracy_score(y_pred=y_prediction, y_true=y_test)
```
leads to different results as well.  It looks like the model either uses some random function (AFAIK all: ```random_states=42``` in the source code) or there is some major error in the pipeline that causes different predictions of the test set all the time.

### Expected Behavior
Final evaluation on a test set should not lead to different results using the same model on the same data.

### Setup Details
Include the details about the versions of:
 - OS type and version: Linux
 - Python: 3.6.5
 - autokeras: 0.2.11
 - scikit-learn:0.19.1
 - numpy:1.14.5
 - keras: 2.2.2
 - scipy:1.1.0
 - tensorflow: 1.10.0
 - pytorch:0.4.1

### Additional context
This error is verified on MNIST and Fashion-MNIST.

**Comment by jhfjhfj1** [comment]
Thank you so much! @swenkel 
This is a very important bug.
I have just fixed it in the develop branch, which will be used for the next release.

**Comment by JaeDukSeo** [comment]
thank you for fixing this
